foundation-ui/services/MountBundle.js

```diff
@@ -0,0 +1,147 @@
+import React, {PropTypes} from 'react';
+
+import Hooks from './Hooks';
+import ServiceFactory from '../factories/ServiceFactory';
+
+const hooks = new Hooks();
+const METHODS_TO_BIND = [
+  'updateState'
+];
+
+/*
+ * Example usage;
+ * import {Mount, MountService} from 'foundation-ui/services/MountBundle';
+ *
+ * class ServiceDetailPlugin {
+ *   getContents(children, props) {
+ *     return <ServiceDetail {...props} />;
+ *   }
+ *
+ *   initialize() {
+ *     // priority order is from [-Inifinity, Inifinity]
+ *     MountService.on('serviceDetail', this.getContents, 0);
+ *   }
+ * }
+ *
+ * class MyComponent extends React.Component {
+ *   render() {
+ *     return (
+ *       <Mount mountId="serviceDetail">
+ *         // What to show when serviceDetail doesn't mount
+ *         <EmptyPage />
+ *       </Mount>
+ *     );
+ *   }
+ * };
+ */
+class Mount extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getChildContext() {
+    // Get context passed down from parent Mount if one exists
+    // and add to it so we can create a render tree chain of nested mountIds.
+    // This should just be an Array of string mountIds. We can use the mountIds
+    // to look up each Mount Component on the page and see debug info about it
+    // like which package inserted the component, what the priorities were for
+    // each component if multiple components were registered for a mountId etc.
+    const {mountChain = []} = this.context || {};
+
+    return {mountChain: [...mountChain, this.props.mountId]};
```

Comment:
We talked offline about this. What was meant was a suggestion to use JavaScript Symbols, so we can differentiate between siblings with the same key.

However, we decided that this will work for now and we can give it another look once we implement the debugging tools